BootloaderCommonPkg/Library/TpmLib/TpmEventLog.c

```diff
@@ -382,4 +382,51 @@ TpmLogLocalityEvent (
   }
 
   TpmLogEvent ( &PcrEventHdr, (const UINT8 *)&StartupLocalityEvent);
+}
+
+/**
+  Get the TCG log Lasa and LasaEven info.
+
+  @param[out] Tcg2Lasa  Pointer to the Log area start address .
+  @param[out] Tcg2LasaEven  Pointer to the Last event logged.
+
+**/
+VOID
+GetTCGLogInfo (
+  OUT  UINT32  *Tcg2Lasa,
+  OUT  UINT32  *Tcg2LasaEven
+  )
+{
+
+  UINT32                Lasa;            //LogAreaStartAddress
+  UINT32                Laml;            //LogAreaMinimumLength
+  UINT32                EventSize;
+  TCG_PCR_EVENT2_HDR    *EmptySlot;
+  TCG_PCR_EVENT_HDR     *FirstEvent;
+
+  GetTCGLasa (&Lasa, &Laml);
+
+  if (Lasa == 0 || Laml == 0 ) {
+    DEBUG ((DEBUG_WARN, "Unable to get log area for TCG 2.0 format events !!\n"));
+    *Tcg2Lasa = 0;
```

Comment:
Is this function an external interface ?  If so, we need to check if Tcg2Lasa/Tcg2LasaEven is NULL.

Reply:
Yes, its an external interface. I would add NULL check for Tcg2Lasa and Tcg2LasaEven.